### PR TITLE
Added DHF4 project with test cases for verifying the deployment

### PR DESCRIPTION
**Pull request by rjrudin**
Most of the tests in VerifyDeploymentTest do not pass when using DHF 4.0.3.

I figure we can add to this too as we identify additional resources that a user is likely to deploy in a DHF app.

### CODE REVIEW

File: examples/dhf4-with-tests/build.gradle [hunk 1]
```diff
@@ -0,0 +1,40 @@
+buildscript {
+    repositories {
+        jcenter()
+        maven {url 'http://distro.marklogic.com/nexus/repository/maven-releases/'}
+    }
+    dependencies {
+        classpath "com.marklogic:ml-data-hub:4.1.0-rc1"
+    }
```
Comment (srinathgit) [inline examples/dhf4-with-tests/build.gradle]: I guess this will have to be changed to 4.1.0 after release

Comment (rjrudin) [reply examples/dhf4-with-tests/build.gradle]: Yeah I just marked this as WIP - I'm going to update it as I test out new bug fixes.

**Comment by rjrudin** [comment]
@srinathgit  @akshaysonvane  I verified that this is running fine with 4.1.0-rc2 and all the tests pass. We can still wait on merging it until 4.1.0 is released.

**Comment by srinathgit** [comment]
Thanks @rjrudin for verifying that it is running fine with 4.1.0-rc2.

**Comment by aebadirad** [comment]
@rjrudin Bring this in, or wait for the changes to the tests you're working on?

**Comment by rjrudin** [comment]
@aebadirad  @srinathgit  @akshaysonvane I updated this to depend on 4.1.0. I retested it locally, all looks fine. Also ran mlReloadModules and then ran the test cases, everything looked good.

**Comment by aebadirad** [comment]
Alright. Bringing this in.

**Pull request merged**